### PR TITLE
Format duration for media

### PR DESCRIPTION
**Pull request by sangwoo108**
image
image

Resolves https://github.com/brave/brave-browser/issues/32116

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/playlist/browser/resources/components/header.tsx [hunk 1]
```diff
@@ -88,8 +88,7 @@ export default function Header ({ playlistId }: HeaderProps) {
   const playlist = usePlaylist(playlistId)
 
   const contextualMenuItems = playlist
-    ? []
```
Comment (fallaciousreasoning) [inline components/playlist/browser/resources/components/header.tsx:91]: :laughing:

**Review by fallaciousreasoning — APPROVED** [review]
LGTM!

File: components/playlist/browser/resources/utils/timeFormatter.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import { PlaylistItem } from 'gen/brave/components/playlist/common/mojom/playlist.mojom.m.js'
+
+export enum DelimType {
```
[the remark this answers is not in the document]
Comment (sangwoo108) [reply components/playlist/browser/resources/utils/timeFormatter.ts]: Brilliant! That makes sense. Thanks a lot for the great tip!

File: components/playlist/browser/resources/utils/timeFormatter.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      parts.unshift(stringFormat(hours))
+    }
+    return parts.join(':')
+  }
+
+  const parts = []
+  if (hours) parts.push(`${hours}h`)
+  if (minutes) parts.push(`${minutes}m`)
+  if (seconds) parts.push(`${seconds}s`)
+  return parts.join(' ')
```
[the remark this answers is not in the document]
Comment (sangwoo108) [reply components/playlist/browser/resources/utils/timeFormatter.ts:40]: It's definately better!

File: components/playlist/browser/resources/components/playlistItem.tsx [hunk 4]
```diff
@@ -108,7 +114,11 @@ export default function PlaylistItem ({
         <PlaylistItemName>{name}</PlaylistItemName>
         {
           <ItemDetailsContainer>
-            {<span>duration</span> && false /* TODO(sko) not ready yet */}
+            {duration && (
```
[the remark this answers is not in the document]
Comment (sangwoo108) [reply components/playlist/browser/resources/components/playlistItem.tsx:116]: Yeah, it's string. I think I made the wrong call when working on mojom 🥲

File: components/playlist/browser/resources/components/playlistItem.tsx [hunk 5]
```diff
@@ -65,10 +68,13 @@ const ItemDetailsContainer = styled.div`
...
-  id,
-  name,
-  thumbnailUrl,
-  cached,
+  item: {
+    id,
+    name,
+    cached,
+    duration,
+    thumbnailPath: { url: thumbnailUrl }
```
[the remark this answers is not in the document]
Comment (sangwoo108) [reply components/playlist/browser/resources/components/playlistItem.tsx:75]: Yeah, it took quite a long time for me to take in the syntax 😵‍💫

File: components/playlist/browser/resources/components/playlistInfo.tsx [hunk 6]
```diff
@@ -24,6 +25,7 @@ export const PlaylistInfoContainer = styled.div`
   grid-template-rows: repeat(2, auto);
   grid-template-columns: repeat(3, auto);
   column-gap: 8px;
+  justify-content: left;
```
[the remark this answers is not in the document]
Comment (sangwoo108) [reply components/playlist/browser/resources/components/playlistInfo.tsx]: Nope, `start` would be better.

File: components/playlist/browser/resources/components/header.tsx (hunk 1 above)
Comment (sangwoo108) [reply components/playlist/browser/resources/components/header.tsx:91]: Exactly what I was like 😄

File: components/playlist/browser/resources/components/playlistItem.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (fallaciousreasoning) [reply components/playlist/browser/resources/components/playlistItem.tsx:75]: Same, half the time I feel like I need to relearn it :laughing:

**Review by fallaciousreasoning — APPROVED** [review]
Still LGTM!

**Pull request merged**